plugin/pkg/scheduler/generic_scheduler.go

```diff
@@ -138,6 +139,15 @@ func findNodesThatFit(pod *api.Pod, podLister algorithm.PodLister, predicateFunc
 			filtered = append(filtered, node)
 		}
 	}
+	if len(filtered) > 0 && len(extenders) != 0 {
+		for _, extender := range extenders {
+			filteredList, err := extender.Filter(pod, &api.NodeList{Items: filtered})
```

Comment:
Yeah, this is a really painful decision. Without passing the list of pods running on each machine, the extender writer is really limited in what they can do. For example, they can't do anything that requires knowing the free and used resources on the machine (since that requires knowing the requests of the pods on the machine), or implement a policy that avoids conflicts between the pod-to-schedule and already-running pods. OTOH I think the whole mechanism will be impossibly slow if we pass the list of pods on each machine. This is why this mechanism is really a bad way to extend the scheduler. I wish you would consider some other approach, like figuring out a way we can dynamically link scheduling policies (predicates and priorities) into the existing scheduler. (Dynamically link meaning requires restarting the scheduler process but not recompiling.) Since we already agreed to let you do this, I'm going to say what you have here is OK, but I have a feeling you're the only person who is ever going to use this extension mechanism.